backend/src/main/java/eu/bbmri_eric/negotiator/TemplateInitiationRunner.java

```diff
@@ -0,0 +1,64 @@
+package eu.bbmri_eric.negotiator;
+
+import eu.bbmri_eric.negotiator.notification.UserNotificationService;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.PathMatchingResourcePatternResolver;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TemplateInitiationRunner implements CommandLineRunner {
+
+  private final UserNotificationService userNotificationService;
+
+  @Value("${spring.thymeleaf.prefix:classpath:/templates/}")
+  private String thymeleafPrefix;
+
+  @Value("${spring.thymeleaf.suffix:.html}")
+  private String thymeleafSuffix;
+
+  private String defaultTemplatePath = "classpath:/templates/";
+
+  @Autowired
+  public TemplateInitiationRunner(UserNotificationService userNotificationService) {
+    this.userNotificationService = userNotificationService;
+  }
+
+  @Override
+  public void run(String... args) throws Exception {
```

Comment:
this code is way to complicated, we should divide it into multiple functions

Reply:
split it in multiple functions